### PR TITLE
#212 fixed warning from PyCharm on FromComponent

### PR DESCRIPTION
**Pull request by IvanKirpichnikov**

### CODE REVIEW

**Comment by Tishka17** [comment]
Let's keep it open for a while in case Pycharm developers fix it on their end

**Comment by Forceres** [comment]
> Let's keep it open for a while in case Pycharm developers fix it on their end

These warnings have kept on several month yet. I don't think Pycharm devs are planning to fix it

**Comment by trancong12102** [comment]
@Forceres I also think JetBrains won't fix this issue soon. For example, this issue has existed for more than 2 years.
https://youtrack.jetbrains.com/issue/PY-54560/Support-PEP-681-typing.dataclasstransform

**Comment by IvanKirpichnikov** [comment]
I'll give you another example. https://youtrack.jetbrains.com/issue/PY-20927/Support-standard-types-for-AsyncIterable-and-AsyncIterator-from-typing-and-collections.abc

File: src/dishka/entities/key.py [hunk 1]
```diff
@@ -5,8 +5,15 @@
 from .component import DEFAULT_COMPONENT, Component
 
 
-class FromComponent(NamedTuple):
-    component: Component = DEFAULT_COMPONENT
+class _FromComponent(NamedTuple):
```
Comment (RonnyPfannschmidt) [inline src/dishka/entities/key.py:8]: how about using a frozen dataclass or a concrete class with a ctor, this isnt exactly a actual tuple in use, its more like a stash type to add a annotation

**Pull request closed without merging**